### DISCUSSION TITLE
Can not access state vector values within model_interpolate

### DISCUSSION THREAD

**Discussion by sr-dash** [NCAR/DART#581] [Q&A]
I am using DART-9.16.4 to intergate my surface flux transport model and reconstruct a model variable by assimilating magnetic field values. In order to generate expected_obs for filter stage I should have access to all the state vector values if ens_size = 20, then 20 values of the state vector at that iteration. Within the model_interpolate routine If I have the values of e. g. 20 states, then I can pass them to my srface flux transport model for that time instance and generate 20 expected observations. This would be passed to expected_obs array for assimilation within the DART framework. 

Can anone suggest a suitable subroutine or function to achieve the same?

Thanks a lot in advance.

**Comment by hkershaw-brown** [comment]
moved to email support.
